tests/testing-resources/include/aws/testing/mocks/aws/client/MockAWSClient.h

```diff
@@ -97,12 +105,21 @@ class MockAWSErrorMarshaller : public Aws::Client::AWSErrorMarshaller
 class MockAWSClient : Aws::Client::AWSClient
 {
 public:
-    MockAWSClient(const Aws::Client::ClientConfiguration& config) : AWSClient(config,
-            Aws::MakeShared<Aws::Client::AWSAuthV4Signer>("MockAWSClient",
-                Aws::MakeShared<Aws::Auth::SimpleAWSCredentialsProvider>("MockAWSClient", GetMockAccessKey(),
+
+     MockAWSClient(const Aws::Client::ClientConfiguration& config, 
+                  const std::shared_ptr<Aws::Client::AWSErrorMarshaller>& errorMarshaller
+    ) : AWSClient(  config,
+                    Aws::MakeShared<Aws::Client::AWSAuthV4Signer>("MockAWSClient",
+                    Aws::MakeShared<Aws::Auth::SimpleAWSCredentialsProvider>("MockAWSClient", GetMockAccessKey(),
                     GetMockSecretAccessKey()), "service", config.region.empty() ? Aws::Region::US_EAST_1 : config.region),
-            Aws::MakeShared<MockAWSErrorMarshaller>("MockAWSClient")),
-        m_countedRetryStrategy(std::static_pointer_cast<CountedRetryStrategy>(config.retryStrategy)) { }
+                    errorMarshaller
```

Comment:
Just adding delegate constructor and leveraging it in passing different error marshallers for different test use cases